### PR TITLE
LPS-170171 | master

### PR DESCRIPTION
**Pull request by sontruongces**
Hi team,

This is the fix for the [LPS-170171](https://issues.liferay.com/browse/LPS-170171), the original has passed my review at sontruongces#67. Please help me to review and give feedback.

Thanks!
cc: @TaiPhamLR

### CODE REVIEW

**Comment by liferay-lima** [comment]
ci:test:sf

**Comment by liferay-lima** [comment]
ci:test:relevant

File: modules/apps/knowledge-base/knowledge-base-web/src/main/resources/META-INF/resources/display/view.jsp [hunk 1]
```diff
@@ -46,10 +46,19 @@ request.setAttribute(KBWebKeys.KNOWLEDGE_BASE_KB_NAVIGATION_DISPLAY_CONTEXT, kbN
...
 		renderRequest.setAttribute(KBWebKeys.PORTLET_CONFIGURATOR_VISIBILITY, Boolean.TRUE);
+
+		String messageText = null;
+
+		if (KBFolderConstants.DEFAULT_PARENT_FOLDER_ID != kbDisplayPortletInstanceConfiguration.resourcePrimKey()) {
+			messageText = "the-selected-knowledge-base-is-empty";
+		}
+		else {
+			messageText = "please-configure-this-portlet-to-make-it-visible-to-all-users";
+		}
```
Comment (adolfopa) [inline modules/apps/knowledge-base/knowledge-base-web/src/main/resources/META-INF/resources/display/view.jsp:57]: Can you add this logic as a new `getAlertMessageLabel()` method in the KBNavigationDisplayContext class?

**Review by adolfopa — CHANGES_REQUESTED** [review]

File: modules/apps/knowledge-base/knowledge-base-web/src/main/resources/META-INF/resources/display/view.jsp (hunk 1 above)
Comment (TaiPhamLR) [reply modules/apps/knowledge-base/knowledge-base-web/src/main/resources/META-INF/resources/display/view.jsp]: Thanks for your feedback @adolfopa. I fixed it.

File: modules/apps/knowledge-base/knowledge-base-web/src/main/java/com/liferay/knowledge/base/web/internal/display/context/KBNavigationDisplayContext.java [hunk 2]
```diff
@@ -55,6 +55,22 @@ public KBNavigationDisplayContext(
...
+
+			messageText = "the-selected-knowledge-base-is-empty";
+		}
+		else {
+			messageText =
+				"please-configure-this-portlet-to-make-it-visible-to-all-users";
+		}
+
+		return messageText;
+	}
```
Comment (adolfopa) [inline modules/apps/knowledge-base/knowledge-base-web/src/main/java/com/liferay/knowledge/base/web/internal/display/context/KBNavigationDisplayContext.java:72]: Can you return the value directly?
```
if (KBFolderConstants.DEFAULT_PARENT_FOLDER_ID != ...) {
     return "the-selected...";
}
else {
   return ...;
}
```

**Review by adolfopa — CHANGES_REQUESTED** [review]

File: modules/apps/knowledge-base/knowledge-base-web/src/main/java/com/liferay/knowledge/base/web/internal/display/context/KBNavigationDisplayContext.java (hunk 2 above)
Comment (TaiPhamLR) [reply modules/apps/knowledge-base/knowledge-base-web/src/main/java/com/liferay/knowledge/base/web/internal/display/context/KBNavigationDisplayContext.java]: I fixed it. Thanks.

**Review by adolfopa — APPROVED** [review]

**Comment by adolfopa** [comment]
ci:forward

**Pull request closed without merging**